### PR TITLE
Code refactoring & added more testcases

### PR DESCRIPTION
**Pull request by hwangswan**
# Code refactoring & added more testcases
## 1. Description
Code refactoring & added more testcases.

## 2. List of issues fixed:
None

Details:
- [x] Added type hinting.
- [x] Added 2 more testcases
- [x] `StudentFactory` now returning Vietnamese (Unicode) in `vn` language.

Mentioning @trhgquan for review

### CODE REVIEW

File: tests/factory/student_factory.py [hunk 1]
```diff
@@ -30,30 +43,30 @@ def __init__(self):
...
-            "G": GraduationType(name_vn="Gioi", name_en="Very good"),
-            "KH": GraduationType(name_vn="Kha", name_en="Good"),
-            "TBK": GraduationType(name_vn="Trung binh kha", name_en="Average good"),
-            "TB": GraduationType(name_vn="Trung binh", name_en="Average")
+            "XS": GraduationType(name_vn="Xuất sắc", name_en="Excellent"),
+            "G": GraduationType(name_vn="Giỏi", name_en="Very good"),
+            "KH": GraduationType(name_vn="Khá", name_en="Good"),
+            "TBK": GraduationType(name_vn="Trung bình khá", name_en="Average good"),
+            "TB": GraduationType(name_vn="Trung bình", name_en="Average")
         }
```
Comment (trhgquan) [inline tests/factory/student_factory.py:67]: Should be mentioned in PR

**Review by trhgquan — CHANGES_REQUESTED** [review]
Requesting changes. Please have a look at comments.

File: tests/factory/student_factory.py (hunk 1 above)
Comment (hwangswan) [reply tests/factory/student_factory.py:67]: Affirmative.

**Review by trhgquan — APPROVED** [review]
**LGTM**

**Pull request merged**